torchao/quantization/quantize_/workflows/float8/float8_tensor.py

```diff
@@ -0,0 +1,618 @@
+        act_quant_kwargs (QuantizeTensorToFloat8Kwargs): the kwargs for Float8Tensor.to_float8
+        general_kernel_preference (GeneralKernelPreference): the preference for quantize, mm etc. kernel to use,
+        by default, this will be chosen for user, for float8 dynamic activation and float8 weight quantization, in SM89, we'll use (`torch._scaled_mm`) and in SM90+ we'll use (`torch.ops.fbgemm.f8f8bf16_rowwise`)
+        dtype: Original Tensor dtype
+    """
+
+    tensor_data_names = ["qdata", "scale"]
+    tensor_attribute_names = [
+        "block_size",
+        "mm_config",
+        "hp_value_lb",
+        "hp_value_ub",
+        "act_quant_kwargs",
+        "general_kernel_preference",
+        "dtype",
+    ]
+
+    def __new__(
+        cls,
+        qdata,
+        scale,
+        block_size,
+        mm_config,
+        hp_value_lb,
+        hp_value_ub,
+        act_quant_kwargs,
+        general_kernel_preference,
+        dtype,
+    ):
+        shape = qdata.shape
+        kwargs = {}
+        kwargs["device"] = qdata.device
+        kwargs["dtype"] = dtype
+        kwargs["requires_grad"] = False
+        return torch.Tensor._make_wrapper_subclass(cls, shape, **kwargs)  # type: ignore[attr-defined]
+
+    def __init__(
+        self,
+        qdata: torch.Tensor,
+        scale: torch.Tensor,
+        block_size: Optional[List[int]] = None,
+        mm_config: Optional[Float8MMConfig] = None,
+        hp_value_lb: Optional[float] = None,
+        hp_value_ub: Optional[float] = None,
+        act_quant_kwargs: Optional[QuantizeTensorToFloat8Kwargs] = None,
+        general_kernel_preference: GeneralKernelPreference = GeneralKernelPreference.DEFAULT,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        self.qdata = qdata
+        self.scale = scale
+        self.block_size = block_size
+        self.mm_config = mm_config
+        self.hp_value_lb = hp_value_lb
+        self.hp_value_ub = hp_value_ub
+        self.act_quant_kwargs = act_quant_kwargs
+        self.general_kernel_preference = general_kernel_preference
+
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}({self.act_quant_kwargs=}, {self.qdata=}, {self.scale=}, "
+            f"{self.block_size=}, {self.mm_config=}, "
+            f"{self.shape=}, {self.device=}, {self.dtype=})"
+        )
+
+    def _quantization_type(self):
+        return f"{self.act_quant_kwargs=}, {self.block_size=}, {self.mm_config=}, {self.scale.shape=}"
+
+    def dequantize(self, output_dtype: Optional[torch.dtype] = None) -> torch.Tensor:
+        if output_dtype is None:
+            output_dtype = self.dtype
+
+        qdata, scale = self.qdata, self.scale
+        return _dequantize_affine_float8(qdata, scale, output_dtype)
+
+    @classmethod
+    def to_float8(
+        cls,
+        hp_tensor: torch.Tensor,
+        quant_kwargs: QuantizeTensorToFloat8Kwargs,
```

Comment:
TBH I'd prefer to spell out the weight arguments here, and only wrap the activation arguments in `QuantizeTensor...Kwargs`.  I know it's duplication, but IMO the code would be easier to follow for the common case and I like that.